rpcs3/Input/pad_thread.cpp

```diff
@@ -40,15 +42,13 @@ pad_thread::pad_thread(void *_curthread, void *_curwindow, std::string_view titl
 	pad::g_title_id = title_id;
 	Init();
 
-	thread = std::make_shared<std::thread>(&pad_thread::ThreadFunc, this);
 	pad::g_current = this;
 }
 
 pad_thread::~pad_thread()
 {
 	pad::g_current = nullptr;
 	pad::g_active = false;
-	thread->join();
```

Comment:
Nothing stops you from joining the thread and it is needed, you need to assign thread_state::finished to it.

Reply:
why, we don't do it for any fxo thread

Reply:
Or actually just remove the following line `handlers.clear()` because of thread-safety issues.

Reply:
done